greenmail-core/src/main/java/com/icegreen/greenmail/user/GreenMailUser.java

```diff
@@ -28,4 +28,6 @@ public interface GreenMailUser {
 
     String getQualifiedMailboxName();
 
+    void setTokenValidator(TokenValidator validator);
```

Comment:
Would it be possible to keep this setter out of GreenMailUser interface, and to only keep the method in the UserImpl for now?

Idea would be to hide behind the `authenticate(...)` method. The whole GreenMail auth part is very grown/legacy-heavy and needs some rework in the future...

Reply:
Sure, will do. If the refactoring can be described somehow lets tackle it in another issue? For now I can do as suggested.